odoo/addons/base/models/res_config.py

```diff
@@ -451,12 +447,18 @@ def _get_classified_fields(self):
                     'other':   ['other_field', ...],
                 }
         """
-        IrModule = self.env['ir.module.module']
+        IrModule = self.env['ir.module.module'].sudo()
+        IrModelData = self.env['ir.model.data']
         Groups = self.env['res.groups']
-        ref = self.env.ref
+
+        def ref(xml_id):
+            model, id = IrModelData.xmlid_to_res_model_res_id(xml_id)
+            return self.env[model].browse(id)
+        fnames = fnames or self._fields.keys()
```

Comment:
Maybe check if `fnames is None`? I could imagine a situation where the caller passes in an empty list of `fnames` and could be very surprised to receive a bunch of garbage.